From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Fri, 22 Oct 2010 09:51:46 -0400
Subject: [virt] xen: increase txqueuelen of netback vif devices
Message-id: <331199966.114651287741106821.JavaMail.root@zmail06.collab.prod.int.phx2.redhat.com>
Patchwork-id: 28897
O-Subject: [RHEL5.6 kernel-xen PATCH] Increase txqueuelen of vif devices
Bugzilla: 539626
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Michal Novotny <minovotn@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>

BZ 539626 - default txqueuelen of vif device is too smal
https://bugzilla.redhat.com/show_bug.cgi?id=539626

Xen uses txqueuelen == 32 for vif devices (netback devices). This value kills through put
of devices. Increasing this value can significantly increase throughput. We are going to set
this value to 500 as this value is used by kvm. There's possibility to change this default
using module parameter queue_length.

Patch:
--

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/xen/netback/interface.c b/drivers/xen/netback/interface.c
index b475006..33f3315 100644
--- a/drivers/xen/netback/interface.c
+++ b/drivers/xen/netback/interface.c
@@ -49,7 +49,7 @@
  * run a timer (tx_queue_timeout) to drain the queue when the interface is
  * blocked.
  */
-static unsigned long netbk_queue_length = 32;
+static unsigned long netbk_queue_length = 500;
 module_param_named(queue_length, netbk_queue_length, ulong, 0);
 
 static void __netif_up(netif_t *netif)
